app/controllers/ems_common.rb

```diff
@@ -571,12 +571,15 @@ def recheck_authentication(id = nil)
   end
 
   def check_compliance(model)
-    emss = find_checked_items
-    if emss.empty?
-      add_flash(_("No %{model} were selected for Compliance Check") % {:model => ui_lookup(:models => model.to_s)}, :error)
+    showlist = @lastaction == "show_list"
+    ids = showlist ? find_checked_items : ([params[:id]] if model.find(params[:id]))
```

Comment:
@enoodle do we really need a model.find call here, all we need to make sure is that a record or multiple records from list were selected, so 'ids' shouldn't be blank.